src/main/java/com/coveros/selenified/utilities/Listener.java

```diff
@@ -55,90 +58,111 @@
     /**
      * determines the folder name associated with the given tests
      *
-     * @param test - the testng itestresult object
+     * @param result - the testng itestresult object
      * @return String: a string version of the folder name
      */
-    private static String getFolderName(ITestResult test) {
-        return new File(test.getTestContext().getOutputDirectory()).getName();
+    private static String getFolderName(ITestResult result) {
+        return new File(result.getTestContext().getOutputDirectory()).getName();
     }
 
     /**
      * determines the test name associated with the given tests
      *
-     * @param test - the testng itestresult object
+     * @param result - the testng itestresult object
      * @return String: a string version of the test name
      */
-    private static String getTestName(ITestResult test) {
+    private static String getTestName(ITestResult result) {
         String className;
         String packageName = "";
-        if (test.getTestClass().toString().contains(".")) {
-            packageName = test.getTestClass().toString().substring(22, test.getTestClass().toString().length() - 1)
+        if (result.getTestClass().toString().contains(".")) {
+            packageName = result.getTestClass().toString().substring(22, result.getTestClass().toString().length() - 1)
                     .split("\\.")[0];
-            className = test.getTestClass().toString().substring(22, test.getTestClass().toString().length() - 1)
+            className = result.getTestClass().toString().substring(22, result.getTestClass().toString().length() - 1)
                     .split("\\.")[1];
         } else {
-            className = test.getTestClass().toString().substring(22, test.getTestClass().toString().length() - 1);
+            className = result.getTestClass().toString().substring(22, result.getTestClass().toString().length() - 1);
+        }
+        return TestSetup.getTestName(packageName, className, result.getName(), result.getParameters());
+    }
+
+    /**
+     * Provides ability to skip a test, based on the browser selected
+     * @param result - the testng itestresult object
+     */
+    @Override
+    public void onTestStart(ITestResult result) {
+        super.onTestStart(result);
+        // if a group indicates an invalid browser, skip the test
+        Browser browser = (Browser) result.getAttribute(BROWSER_INPUT);
+        if (browser != null) {
+            String[] groups = result.getMethod().getGroups();
+            for (String group : groups) {
+                if (group.toLowerCase().equals("no-" + browser.getName().toString().toLowerCase())) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace these toUpperCase()/toLowerCase() and equals() calls with a single equalsIgnoreCase() call. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1157)